subsys/disk/disk_access.c

```diff
@@ -0,0 +1,179 @@
+		 * name is shorter than the mount point name.
+		 */
+		if (strlen(itr->name) != name_len) {
+			continue;
+		}
+
+		/* Check for disk name match */
+		if (strncmp(name, itr->name, name_len) == 0) {
+			disk = itr;
+		}
+	}
+	k_mutex_unlock(&mutex);
+
+	return disk;
+}
+
+int disk_access_init(const char *pdrv)
+{
+	struct disk_info *disk = disk_access_get_di(pdrv);
+	int rc = -EINVAL;
+
+	if ((disk != NULL) || (disk->ops != NULL) ||
+				(disk->ops->init != NULL)) {
+		rc = disk->ops->init(disk);
+	}
+
+	return rc;
+}
+
+int disk_access_status(const char *pdrv)
+{
+	struct disk_info *disk = disk_access_get_di(pdrv);
+	int rc = -EINVAL;
+
+	if ((disk != NULL) || (disk->ops != NULL) ||
+				(disk->ops->status != NULL)) {
+		rc = disk->ops->status(disk);
+	}
+
+	return rc;
+}
+
+int disk_access_read(const char *pdrv, u8_t *data_buf,
+		     u32_t start_sector, u32_t num_sector)
+{
+	struct disk_info *disk = disk_access_get_di(pdrv);
+	int rc = -EINVAL;
+
+	if ((disk != NULL) || (disk->ops != NULL) ||
+				(disk->ops->read != NULL)) {
+		rc = disk->ops->read(disk, data_buf, start_sector, num_sector);
+	}
+
+	return rc;
+}
+
+int disk_access_write(const char *pdrv, const u8_t *data_buf,
+		      u32_t start_sector, u32_t num_sector)
+{
+	struct disk_info *disk = disk_access_get_di(pdrv);
+	int rc = -EINVAL;
+
+	if ((disk != NULL) || (disk->ops != NULL) ||
+				(disk->ops->write != NULL)) {
+		rc = disk->ops->write(disk, data_buf, start_sector, num_sector);
+	}
+
+	return rc;
+}
+
+int disk_access_ioctl(const char *pdrv, u8_t cmd, void *buf)
+{
+	struct disk_info *disk = disk_access_get_di(pdrv);
+	int rc = -EINVAL;
+
+	if ((disk != NULL) || (disk->ops != NULL) ||
+				(disk->ops->ioctl != NULL)) {
+		rc = disk->ops->ioctl(disk, cmd, buf);
+	}
+
+	return rc;
+}
+
+int disk_access_register(struct disk_info *disk)
+{
+	int rc = 0;
+
+	k_mutex_lock(&mutex, K_FOREVER);
+	if ((disk == NULL) || (disk->name == NULL)) {
+		SYS_LOG_ERR("invalid disk interface!!");
+		rc = -EINVAL;
+		goto reg_err;
+	}
+
+	if (disk_access_get_di(disk->name) != NULL) {
+		SYS_LOG_ERR("disk interface already registered!!");
+		rc = -EINVAL;
+		goto reg_err;
+	}
+
+	/*  append to the disk list */
+	sys_dlist_append(&disk_access_list, &disk->node);
+	SYS_LOG_DBG("disk interface(%s) registred", name);
+reg_err:
+	k_mutex_unlock(&mutex);
+	return rc;
+}
+
+int disk_access_unregister(struct disk_info *disk)
+{
+	int rc = 0;
+
+	k_mutex_lock(&mutex, K_FOREVER);
+	if ((disk == NULL) || (disk->name == NULL)) {
+		SYS_LOG_ERR("invalid disk interface!!");
+		rc = -EINVAL;
+		goto unreg_err;
+	}
+
+	if (disk_access_get_di(disk->name) == NULL) {
+		SYS_LOG_ERR("disk interface not registered!!");
+		rc = -EINVAL;
+		goto unreg_err;
+	}
+	/* remove disk node from the list */
+	sys_dlist_remove(&disk->node);
+	SYS_LOG_DBG("disk interface(%s) unregistred", name);
+unreg_err:
+	k_mutex_unlock(&mutex);
+	return rc;
+}
+
+static int disk_init(struct device *dev)
```

Comment:
If the mutex and the list are initalized statically, you can get rid of this function

Reply:
I would like to still have disk_init as placeholder for future purpose.